src/go/cmd/k8s-api-test/main.go

```diff
@@ -13,32 +13,47 @@
 package main
 
 import (
+	"flag"
+	"fmt"
 	"log"
+	"os"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 )
 
+var help bool
+
+func init() {
+	flag.BoolVar(&help, "help", false, "show help")
+}
+
 func main() {
-    handleErr := func(message string, err error) {
+	flag.Parse()
+	if help {
+		fmt.Println("k8s-api-test is used to test if k8s cluster if working correctly")
+		os.Exit(0)
+	}
+
+	handleErr := func(message string, err error) {
 		if err != nil {
 			log.Fatalf("%s: %v\n", message, err)
 		}
-    }
+	}
 
-    config, err := rest.InClusterConfig()
-    handleErr("Failed when creating in-cluster K8S config", err)
+	config, err := rest.InClusterConfig()
+	handleErr("Failed when creating in-cluster K8S config", err)
 
-    clientset, err := kubernetes.NewForConfig(config)
-    handleErr("Failed when connecting to K8S cluster", err)
+	clientset, err := kubernetes.NewForConfig(config)
+	handleErr("Failed when connecting to K8S cluster", err)
 
-    version, err := clientset.Discovery().ServerVersion()
-    handleErr("Failed when pulling version info", err)
-    log.Printf("Kubernetes version: %s", version.GitVersion)
+	version, err := clientset.Discovery().ServerVersion()
+	handleErr("Failed when pulling version info", err)
+	log.Printf("Kubernetes version: %s", version.GitVersion)
 
-    pods, err := clientset.CoreV1().Pods("").List(metav1.ListOptions{Limit: 20})
-    handleErr("Failed when fetching list of pods", err)
+	pods, err := clientset.CoreV1().Pods("").List(metav1.ListOptions{Limit: 20})
+	handleErr("Failed when fetching list of pods", err)
 
-    log.Printf("Received data for %d pods in the cluster\n", len(pods.Items))
+	log.Printf("Received data for %d pods in the cluster\n", len(pods.Items))
 }
```

Comment:
What is the desire format in terms of spaces vs tabs?

Reply:
I'd say https://golang.org/doc/effective_go.html#formatting

tldr: use tabs

Reply:
This file already used tabs (on the top of the file for instance) in some places and spaces elsewhere.